examples/investigate_streams.rs

```diff
@@ -0,0 +1,266 @@
+                // files, so we call `brotli` manually to obtain one and evaluate
+                // the file size.
+                let brotli_size = get_brotli_size(&path);
+
+                let name = match extension.as_str() {
+                    // For `binjs` and `js` files, we only care about the extension.
+                    "binjs" | "js" => extension.clone(),
+                    _ => path.file_name().unwrap().to_str().unwrap().to_string(),
+                };
+
+                let info = stats
+                    .entry(extension.clone())
+                    .or_insert_with(|| HashMap::new())
+                    .entry(name)
+                    .or_insert_with(|| Compressions::default());
+                info.raw += meta_data.len();
+                info.brotli += brotli_size as u64;
+
+                if options.compare_to_bzip {
+                    info.bzip += get_bzip2_size(&path);
+                };
+                if options.compare_to_lzma {
+                    info.lzma += get_lzma_size(&path);
+                };
+            }
+            // Brotli files, output by binjs_encode.
+            // Remove the `bro` from the name, store in `Compressions::brotli`.
+            "bro" => {
+                let path = path.as_path().with_extension("");
+                let name = path.file_name().unwrap().to_str().unwrap().to_string();
+                let extension = path.extension().unwrap().to_str().unwrap().to_string();
+                let info = stats
+                    .entry(extension.clone())
+                    .or_insert_with(|| HashMap::new())
+                    .entry(name)
+                    .or_insert_with(|| Compressions::default());
+                info.brotli += meta_data.len();
+            }
+            _ => {
+                let name = path.file_name().unwrap().to_str().unwrap().to_string();
+                let info = stats
+                    .entry(extension.clone())
+                    .or_insert_with(|| HashMap::new())
+                    .entry(name)
+                    .or_insert_with(|| Compressions::default());
+                info.raw += meta_data.len();
+                if options.compare_to_bzip {
+                    info.bzip += get_bzip2_size(&path);
+                };
+                if options.compare_to_lzma {
+                    info.lzma += get_lzma_size(&path);
+                };
+            }
+        };
+    }
+
+    eprintln!("Walked {} files", number_of_files);
+
+    // Output results as CSV.
+    // File name / Number of bytes (uncompressed) / Number of bytes (compressed with brotli)
+    print_column(
+        &options,
+        "File",
+        &Compressions {
+            raw: "raw (b)",
+            brotli: "brotli (b)",
+            bzip: "bzip2 (b)",
+            lzma: "lzma (b)",
+        },
+    );
+    let print = |in_extension: &HashMap<String, Compressions<u64>>| {
+        for (name, info) in in_extension.into_iter().sorted().into_iter() {
```

Comment:
Maybe instead of `.sorted()` here it's simpler to use `BTreeMap` in the first place? I imagine performance difference is not important here anyway.

Reply:
Especially since we're sorting ~10 items.

Reply:
Yeah, so `BTreeMap` should be simpler. (Is that what you meant by resolving this conversation?)

Reply:
Oh, I probably misread your previous comment. Yeah, you're right, let's use `BTreeMap`.